diva-irma.js

```diff
@@ -314,6 +315,50 @@ function getIrmaStatus(irmaSessionType, irmaSessionId) {
     });
 }
 
+function jsonQuoteNumberValues(payload) {
+  // Regex for quoting unquoted JSON number values
+  return payload.replace(/: ?(-?\d+)/g, ':"$1"');
+}
+
+function signatureFromToken(token) {
+  // The IRMA signature contains large numerical values. The `jsonwebtoken`
+  // library and even the `JSON.parse` command cannot handle these numeric
+  // values. The token will therefore be split using a custom procedure:
+  // 1. Take the body, not the headers
+  // 2. Decode the JWT, which is base64 encoded
+  // 3. Quote the numeric values for protection
+  // 4. Read the structure with JSON.parse and select the signature
+  // NOTE: As the API can deal with quoted numbers, there is no need to unquote.
```

Comment:
irmago has problems with quoted numbers, but that's not a problem (for now), since we don't connect with irmago.

Reply:
A bit of a risk though. I suggest to fix that before merging it then.

Reply:
Pushed :sunglasses: